build.sh

```diff
@@ -220,188 +232,195 @@ develop)
+  --pull
+  --target main
+  -f "${DOCKERFILE}"
+)
+for IMAGE_NAME in "${IMAGE_NAME_TAGS[@]}"; do
+  DOCKER_BUILD_ARGS+=(-t "${IMAGE_NAME}")
+done
 
-  ###
-  # Proceeding to buils stage, except if `--push-only` is passed
-  ###
-  if [ "${2}" != "--push-only" ]; then
-    ###
-    # Checking if the build is necessary,
-    # meaning build only if one of those values changed:
-    # - base image digest
-    # - netbox git ref (Label: netbox.git-ref)
-    # - netbox-docker git ref (Label: org.opencontainers.image.revision)
-    ###
-    # Load information from registry (only for docker.io)
-    SHOULD_BUILD="false"
-    BUILD_REASON=""
-    if [ -z "${GH_ACTION}" ]; then
-      # Asuming non Github builds should always proceed
-      SHOULD_BUILD="true"
-      BUILD_REASON="${BUILD_REASON} interactive"
-    elif [ "$DOCKER_REGISTRY" = "docker.io" ]; then
-      source ./build-functions/get-public-image-config.sh
-      IFS=':' read -ra DOCKER_FROM_SPLIT <<<"${DOCKER_FROM}"
-      if ! [[ ${DOCKER_FROM_SPLIT[0]} =~ .*/.* ]]; then
-        # Need to use "library/..." for images the have no two part name
-        DOCKER_FROM_SPLIT[0]="library/${DOCKER_FROM_SPLIT[0]}"
-      fi
-      BASE_LAST_LAYER=$(get_image_last_layer "${DOCKER_FROM_SPLIT[0]}" "${DOCKER_FROM_SPLIT[1]}")
-      mapfile -t IMAGES_LAYERS_OLD < <(get_image_layers "${DOCKER_ORG}"/"${DOCKER_REPO}" "${TAG}")
-      NETBOX_GIT_REF_OLD=$(get_image_label netbox.git-ref "${DOCKER_ORG}"/"${DOCKER_REPO}" "${TAG}")
-      GIT_REF_OLD=$(get_image_label org.opencontainers.image.revision "${DOCKER_ORG}"/"${DOCKER_REPO}" "${TAG}")
-
-      if ! printf '%s\n' "${IMAGES_LAYERS_OLD[@]}" | grep -q -P "^${BASE_LAST_LAYER}\$"; then
-        SHOULD_BUILD="true"
-        BUILD_REASON="${BUILD_REASON} debian"
-      fi
-      if [ "${NETBOX_GIT_REF}" != "${NETBOX_GIT_REF_OLD}" ]; then
-        SHOULD_BUILD="true"
-        BUILD_REASON="${BUILD_REASON} netbox"
-      fi
-      if [ "${GIT_REF}" != "${GIT_REF_OLD}" ]; then
-        SHOULD_BUILD="true"
-        BUILD_REASON="${BUILD_REASON} netbox-docker"
-      fi
-    else
-      SHOULD_BUILD="true"
-      BUILD_REASON="${BUILD_REASON} no-check"
-    fi
-    ###
-    # Composing all arguments for `docker build`
-    ###
-    DOCKER_BUILD_ARGS=(
-      --pull
-      --target "${DOCKER_TARGET}"
-      -f "${DOCKERFILE}"
-      -t "${TARGET_DOCKER_TAG}"
-      -t "${TARGET_DOCKER_TAG_PROJECT}"
-    )
-    if [ -n "${TARGET_DOCKER_SHORT_TAG}" ]; then
-      DOCKER_BUILD_ARGS+=(-t "${TARGET_DOCKER_SHORT_TAG}")
-      DOCKER_BUILD_ARGS+=(-t "${TARGET_DOCKER_SHORT_TAG_PROJECT}")
-      DOCKER_BUILD_ARGS+=(-t "${TARGET_DOCKER_LATEST_TAG}")
-      DOCKER_BUILD_ARGS+=(-t "${TARGET_DOCKER_LATEST_TAG_PROJECT}")
-    fi
+# --label
+DOCKER_BUILD_ARGS+=(
+  --label "netbox.original-tag=${TARGET_DOCKER_TAG_PROJECT}"
+  --label "org.opencontainers.image.created=${BUILD_DATE}"
+  --label "org.opencontainers.image.version=${PROJECT_VERSION}"
+)
+if [ -d ".git" ]; then
```

Comment:
`GIT_REF` will not be set if `SKIP_GIT` is true, but a `.git` might still exist. I suggest to change the if and - either additionally or instead – check whether `GIT_REF` is set or not.